server/database/repositories/TournamentRepository.ts

```diff
@@ -1,37 +1,114 @@
+  sideB: MatchSideDB
+}
+function toMatchDb(match: MatchWithoutMS) : MatchDb {
+  const matchDb : MatchDb = {
+    _id: match.id,
+    sideA: {
+      team: toTeamDb(match.sideA.team),
+      score: {
+        a: match.sideA.score.a,
+        b: match.sideA.score.b,
+      },
+    },
+    sideB: {
+      team: toTeamDb(match.sideB.team),
+      score: {
+        a: match.sideB.score.a,
+        b: match.sideB.score.b,
+      },
+    },
+    isFinished: match.isFinished,
+  };
+  return matchDb;
+}
+function toMatch(matchDoc: MatchDocument) : MatchWithoutMS {
+  const match : MatchWithoutMS = {
+    id: matchDoc._id.toString(),
+    sideA: {
+      team: toTeam(matchDoc.sideA.team),
+      score: {
+        a: matchDoc.sideA.score.a,
+        b: matchDoc.sideA.score.b,
+      },
+    },
+    sideB: {
+      team: toTeam(matchDoc.sideB.team),
+      score: {
+        a: matchDoc.sideA.score.a,
+        b: matchDoc.sideA.score.b,
+      },
+    },
+    isFinished: matchDoc.isFinished,
+  };
+  return match;
+}
+interface TournamentDb extends Omit<TournamentWihtoutMS, 'id' | 'teams' | 'matches'> {
+  _id?: string,
+  teams: TeamDb[],
+  matches: MatchDb[]
+}
+function toTournamentDb(t: TournamentWihtoutMS) : TournamentDb {
+  const tDb : TournamentDb = {
+    _id: t.id,
+    name: t.name,
+    ownerId: t.ownerId,
+    teams: t.teams.map((team) => toTeamDb(team)),
+    matches: t.matches.map((match) => toMatchDb(match)),
+  };
+  return tDb;
+}
+function toTournament(tourDoc: TournamentDocument) : TournamentWihtoutMS {
+  const tournament : TournamentWihtoutMS = {
+    id: tourDoc._id.toString(),
+    ownerId: tourDoc.ownerId,
+    name: tourDoc.name,
+    teams: tourDoc.teams.map((teamDoc) => toTeam(teamDoc)),
+    matches: tourDoc.matches.map((matchDoc) => toMatch(matchDoc)),
+  };
+  return tournament;
+}
 
 export default class TournamentRepository {
   public static async create(tournament: TournamentWihtoutMS) {
-    // Check if teams have ids, if teams in matches have ids
-    // and if teams in matches are on teams list
+    if (tournament.id) { throw new AppError('New tournament should not have id', 400); }
     tournament.teams.forEach((team) => {
-      if (!team._id) { throw new AppError('All teams should have ids.', 400); }
+      if (!team.id) { throw new AppError('All teams should have ids.', 400); }
     });
     tournament.matches.forEach((match) => {
-      if (!match.sideA.team._id || !match.sideB.team._id) { throw new AppError('All teams in matches entries should have ids', 400); }
+      if (!match.sideA.team.id || !match.sideB.team.id) { throw new AppError('All teams in matches entries should have ids', 400); }
     });
-
-    return await TournamentModel.create(tournament);
+    const tDoc = await TournamentModel.create(toTournamentDb(tournament));
+    const tour = await TournamentRepository.getById(tDoc._id);
+    if (!tour) { throw new AppError('Error while creating tournament', 400); }
+    return tour;
```

Comment:
`TournamentModel.create` returns tournament, so checking / getting it from database is redundant. Database throw error when it occurs